cli/tools/lint/mod.rs

```diff
@@ -271,18 +272,27 @@ impl WorkspaceLinter {
   ) -> Result<(), AnyError> {
     self.file_count += paths.len();
 
-    let lint_rules = get_config_rules_err_empty(
+    let lint_rules = self.lint_rule_provider.resolve_lint_rules_err_empty(
       lint_options.rules,
       member_dir.maybe_deno_json().map(|c| c.as_ref()),
     )?;
-    let incremental_cache = Arc::new(IncrementalCache::new(
-      self.caches.lint_incremental_cache_db(),
-      &lint_rules.incremental_cache_state(),
-      &paths,
-    ));
+    let maybe_incremental_cache =
+      lint_rules.incremental_cache_state().map(|state| {
+        Arc::new(IncrementalCache::new(
+          self.caches.lint_incremental_cache_db(),
+          &state,
+          &paths,
+        ))
+      });
+
+    let linter = Arc::new(CliLinter::new(CliLinterOptions {
+      configured_rules: lint_rules,
+      fix: lint_options.fix,
+      deno_lint_config: lint_config,
+    }));
```

Comment:
It's most likely a problem for another time, but we're gonna have a lot of "fun" with this abstraction if we start supporting plugins. Again, a problem for another time.

Reply:
I don't think this makes that any harder. Maybe easier.